### DISCUSSION TITLE
How to use deferredLoading for speeding up web

### DISCUSSION THREAD

**Discussion by olegyablokov** [Milad-Akarie/auto_route_library#1694] [Q&A] [answered]
Hi, I am trying to use deferred loading to speed up my webapp which takes ages to load due to lots of pages.

When I add `deferredLoading=true` in `@AutoRouterConfig(...)` parameters and run the generator , it fails with the message:

`Part-file approach will not work with deferred loading because allocator needs to mark all deferred imports!`

I don't know what to do from here. I guess I need to rewrite imports like `import 'package:my_app/presentation/pages/my_page.dart' deferred as my_page;` and remove `part 'app_router.gr.dart';` but after that I need to somehow create `AutoRoute` instances and make everything work...

Could anyone help please? Honestly there is not much on the Internet or in the docs about this particular topic.

**Comment by Milad-Akarie** [comment]
@olegyablokov Just remove the part directive from your router file and re run the generator, you might need to fix some imports after

**Reply by olegyablokov** [reply, marked as answer]
Oh I see! After removing the part directive I need to import the generated file like a normal file: `import 'package:myapp/internal/app_router.gr.dart';`. Also I was importing all my pages in the `app_router.dart` which is unnecessary.

Maybe it's just me but I think it would be a good idea to add this info to README.

Thank you very much!
